tartiflette/sdl/transformers/cleaning_transformer.py

```diff
@@ -0,0 +1,100 @@
+from lark.lexer import Token
+from lark.tree import Transformer_NoRecurse, Tree
+
+from tartiflette.sdl.transformers.helpers import find_token_in_ast
+
+
+class CleaningTransformer(Transformer_NoRecurse):
+    """
+    This lark Transformer cleans up `Token`s and `Tree`s
+    for the SchemaTransformer
+    """
+
+    def __init__(self, input_sdl: str):
+        self.input_sdl = input_sdl
+
+    def name(self, tree: Tree):
+        token = find_token_in_ast(
+            tree.children, [
+                'IDENT',
+                'SCHEMA',
+                'QUERY',
+                'MUTATION',
+                'SUBSCRIPTION',
+                'SCALAR',
+                'TYPE',
+                'INTERFACE',
+                'IMPLEMENTS',
+                'UNION',
+                'ENUM',
+                'INPUT',
+                'EXTEND',
+                'DIRECTIVE',
+                'ON',
+                'TYPE_SYSTEM_DIRECTIVE_LOCATION',
+            ]
+        )
+        token.type = 'IDENT'  # clean up the grammar variations
+        return token
+
+    def description(self, tree: Tree):
+        token = find_token_in_ast(tree.children, ['STRING', 'LONG_STRING'])
+        tmp = bytes(token.value[1:-1], "utf-8").decode('unicode-escape')
```

Comment:
Could you add a small comment regarding `token.value[1:-1]` and `token.value[3:-3]`.

Reply:
`tmp` could be replaced by another name ?

Reply:
Updated 👍

Reply:
Top